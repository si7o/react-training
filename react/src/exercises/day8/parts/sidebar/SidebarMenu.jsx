import React from "react";
import { MenuList, MenuItem } from "@material-ui/core";
import { NavLink } from "react-router-dom";

import SpeedIcon from "@material-ui/icons/Speed";
import SettingsIcon from "@material-ui/icons/Settings";
import RssFeedIcon from "@material-ui/icons/RssFeed";
import RecordVoiceOverIcon from "@material-ui/icons/RecordVoiceOver";
import GroupWorkIcon from "@material-ui/icons/GroupWork";

const SidebarMenu = () => {
  return (
    <MenuList>
      <MenuItem
        component={NavLink}
        to={{
          pathname: "/day8-homework/dashboard",
          state: { title: "Dashboard" },
        }}
        activeClassName="Mui-selected"
      >
        <SpeedIcon /> Dashboard
      </MenuItem>
      <MenuItem
        component={NavLink}
        to={{
          pathname: "/day8-homework/new-campaign",
          state: { title: "New Campaign" },
        }}
        activeClassName="Mui-selected"
      >
        <RecordVoiceOverIcon /> New Campaign
      </MenuItem>
      <MenuItem
        component={NavLink}
        to={{
          pathname: "/day8-homework/integration",
          state: { title: "Integration" },
        }}
        activeClassName="Mui-selected"
      >
        <GroupWorkIcon /> Integration
      </MenuItem>
      <MenuItem
        component={NavLink}
        to={{
          pathname: "/day8-homework/subscribers",
          state: { title: "Subscribers" },
        }}
        activeClassName="Mui-selected"
      >
        <RssFeedIcon /> Subscribers
      </MenuItem>
      <MenuItem
        component={NavLink}
        to={{
          pathname: "/day8-homework/settings",
          state: { title: "Settings" },
        }}
        activeClassName="Mui-selected"
      >
        <SettingsIcon /> Settings
      </MenuItem>
    </MenuList>
  );
};

export default SidebarMenu;
