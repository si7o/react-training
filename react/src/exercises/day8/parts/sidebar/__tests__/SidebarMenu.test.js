import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import SidebarMenu from "../SidebarMenu";
import { Router } from "react-router-dom";
import { createMemoryHistory } from "history";

const memoryHistory = createMemoryHistory();
memoryHistory.push("/day8-homework/dashboard");

describe("<SidebarMenu>", () => {
  test("matches snapshot", () => {
    const { asFragment } = render(
      <Router history={memoryHistory}>
        <SidebarMenu />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("selected item changes", () => {
    const { asFragment } = render(
      <Router history={memoryHistory}>
        <SidebarMenu />
      </Router>
    );
    const menuItems = screen.getAllByRole("menuitem");

    expect(menuItems[0]).toHaveClass("Mui-selected");

    userEvent.click(menuItems[3]);

    expect(menuItems[3]).toHaveClass("Mui-selected");
  });
});
