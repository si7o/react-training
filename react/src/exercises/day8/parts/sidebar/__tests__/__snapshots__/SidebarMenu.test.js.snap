// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarMenu> matches snapshot 1`] = `
<DocumentFragment>
  <ul
    class="MuiList-root MuiList-padding"
    role="menu"
    tabindex="-1"
  >
    <a
      aria-current="page"
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button Mui-selected"
      href="/day8-homework/dashboard"
      role="menuitem"
      tabindex="0"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.38 8.57l-1.23 1.85a8 8 0 01-.22 7.58H5.07A8 8 0 0115.58 6.85l1.85-1.23A10 10 0 003.35 19a2 2 0 001.72 1h13.85a2 2 0 001.74-1 10 10 0 00-.27-10.44zm-9.79 6.84a2 2 0 002.83 0l5.66-8.49-8.49 5.66a2 2 0 000 2.83z"
        />
      </svg>
       Dashboard
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      href="/day8-homework/new-campaign"
      role="menuitem"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="9"
          cy="9"
          r="4"
        />
        <path
          d="M9 15c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4zm7.76-9.64l-1.68 1.69c.84 1.18.84 2.71 0 3.89l1.68 1.69c2.02-2.02 2.02-5.07 0-7.27zM20.07 2l-1.63 1.63c2.77 3.02 2.77 7.56 0 10.74L20.07 16c3.9-3.89 3.91-9.95 0-14z"
        />
      </svg>
       New Campaign
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      href="/day8-homework/integration"
      role="menuitem"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM8 17.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5zM9.5 8c0-1.38 1.12-2.5 2.5-2.5s2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8zm6.5 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5z"
        />
      </svg>
       Integration
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      href="/day8-homework/subscribers"
      role="menuitem"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <circle
          cx="6.18"
          cy="17.82"
          r="2.18"
        />
        <path
          d="M4 4.44v2.83c7.03 0 12.73 5.7 12.73 12.73h2.83c0-8.59-6.97-15.56-15.56-15.56zm0 5.66v2.83c3.9 0 7.07 3.17 7.07 7.07h2.83c0-5.47-4.43-9.9-9.9-9.9z"
        />
      </svg>
       Subscribers
      <span
        class="MuiTouchRipple-root"
      />
    </a>
    <a
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
      href="/day8-homework/settings"
      role="menuitem"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
          transform="scale(1.2, 1.2)"
        />
      </svg>
       Settings
      <span
        class="MuiTouchRipple-root"
      />
    </a>
  </ul>
</DocumentFragment>
`;
